pkg/ssh/ssh.go

```diff
@@ -383,6 +383,9 @@ func (l *SSHTunnelList) removeAndReAdd(e sshTunnelEntry) {
 			l.entries = append(l.entries[:i], l.entries[i+1:]...)
 			l.adding[e.Address] = true
 			go l.createAndAddTunnel(e.Address)
+			if err := e.Tunnel.Close(); err != nil {
```

Comment:
It'd be better to release the lock (`tunnelsLock`) before closing the tunnel. How about grouping the critical section (which mutates `l.entries` and `l.adding`) together so that it's easy to acquire/release the lock, and calls `createAndAddTunnel` and `e.Tunnel.Close()` afterwards?